server/src-lib/Hasura/GraphQL/Context.hs

```diff
@@ -90,378 +24,44 @@ data GCtx
-  -- for now subscription root is query root
-  in GCtx allTys fldInfos ordByEnums queryRoot mutRootM subRootM
-     (Map.map fst flds) insCtxMap
-  where
-    TyAgg tyInfos fldInfos scalars ordByEnums = tyAgg
-    colTys = Set.fromList $ map pgiType $ lefts $ Map.elems fldInfos
-    mkMutRoot =
-      mkHsraObjTyInfo (Just "mutation root") (G.NamedType "mutation_root") Set.empty .
-      mapFromL _fiName
-    mutRootM = bool (Just $ mkMutRoot mFlds) Nothing $ null mFlds
-    mkSubRoot =
-      mkHsraObjTyInfo (Just "subscription root")
-      (G.NamedType "subscription_root") Set.empty . mapFromL _fiName
-    subRootM = bool (Just $ mkSubRoot qFlds) Nothing $ null qFlds
-    (qFlds, mFlds) = partitionEithers $ map snd $ Map.elems flds
     schemaFld = mkHsraObjFldInfo Nothing "__schema" Map.empty $
-                  G.toGT $ G.toNT $ G.NamedType "__Schema"
+                G.toGT $ G.toNT $ G.NamedType "__Schema"
     typeFld = mkHsraObjFldInfo Nothing "__type" typeFldArgs $
-                G.toGT $ G.NamedType "__Type"
-      where
-        typeFldArgs = mapFromL _iviName [
-          InpValInfo (Just "name of the type") "name" Nothing
-          $ G.toGT $ G.toNT $ G.NamedType "String"
-          ]
-
-    anyGeoTypes = any isGeoType colTys
-    allComparableTypes =
-      if anyGeoTypes
-        -- due to casting, we need to generate both geometry and geography
-        -- operations even if just one of the two appears in the schema
-        then Set.union (Set.fromList [PGGeometry, PGGeography]) colTys
-        else colTys
-    allScalarTypes = allComparableTypes <> scalars
-
-    wiredInGeoInputTypes =
-      guard anyGeoTypes *> map TIInpObj
-        [ stDWithinGeometryInputType
-        , stDWithinGeographyInputType
-        , castGeometryInputType
-        , castGeographyInputType
-        ]
-
-    stDWithinGeometryInputType =
-      mkHsraInpTyInfo Nothing stDWithinGeometryInpTy $ fromInpValL
-      [ InpValInfo Nothing "from" Nothing $ G.toGT $ G.toNT $ mkScalarTy PGGeometry
-      , InpValInfo Nothing "distance" Nothing $ G.toNT $ mkScalarTy PGFloat
-      ]
-    stDWithinGeographyInputType =
-      mkHsraInpTyInfo Nothing stDWithinGeographyInpTy $ fromInpValL
-      [ InpValInfo Nothing "from" Nothing $ G.toGT $ G.toNT $ mkScalarTy PGGeography
-      , InpValInfo Nothing "distance" Nothing $ G.toNT $ mkScalarTy PGFloat
-      , InpValInfo
-        Nothing "use_spheroid" (Just $ G.VCBoolean True) $ G.toGT $ mkScalarTy PGBoolean
-      ]
-
-    castGeometryInputType = mkCastExpressionInputType PGGeometry [PGGeography]
-    castGeographyInputType = mkCastExpressionInputType PGGeography [PGGeometry]
+              G.toGT $ G.NamedType "__Type"
+    typeFldArgs = mapFromL _iviName $ pure $
+      InpValInfo (Just "name of the type") "name" Nothing
+      $ G.toGT $ G.toNT $ G.NamedType "String"
 
 emptyGCtx :: GCtx
-emptyGCtx = mkGCtx mempty mempty mempty
+emptyGCtx =
```

Comment:
Previously creating an empty `GCtx` was done with `mkGCtx` which lead to all sorts of cyclic dependencies and code being in incorrect modules. Now `emptyGCtx` is implemented separately with both `mkGCtx` and `emptyGCtx` sharing the common functionality (this could be improved further though).